src/AbstractFloatType.php

```diff
@@ -6,19 +6,23 @@
 use Fortuneglobe\Types\Interfaces\RepresentsFloatValue;
 
 /**
- * Class FloatType
+ * Class AbstractFloatType
  * @package Fortuneglobe\Types
  */
-abstract class FloatType extends AbstractType implements RepresentsFloatValue
+abstract class AbstractFloatType extends AbstractType implements RepresentsFloatValue
 {
 	/** @var float */
 	private $value;
 
 	public function __construct( float $value )
 	{
+		$this->guardValueIsValid( $value );
+
 		$this->value = $value;
 	}
 
+	abstract protected function guardValueIsValid( float $value );
```

Comment:
The return type is omitted.

```php
abstract protected function guardValueIsValid( float $value ): void
```

Regarding to this the guard methods in all abstract types need to be fixed.

Reply:
Fixed in 70fe97374eeed55e6dce6d5d1872fd2f9ec9974c